src/manager.cpp

```diff
@@ -271,15 +271,46 @@ int Manager::updateKeyword(const types::Path i_vpdPath,
         return l_bytesUpdatedOnHardware;
     }
 
-    // Update keyword's value on inventory object path if present
+    // If inventory D-bus object path is present, perform update
     if (!l_inventoryObjPath.empty())
     {
-        int l_bytesUpdatedOnDbus = updateKeywordOnDbus(
-            l_inventoryObjPath, l_fruPath, i_paramsToWriteData);
+        types::Record l_recordName;
+        types::DbusInterface l_interfaceName;
+        types::DbusPropertyName l_propertyName;
+        types::DbusVariantType l_keywordValue;
 
-        if (l_bytesUpdatedOnDbus == -1)
+        if (const types::IpzData* l_ipzData =
+                std::get_if<types::IpzData>(&i_paramsToWriteData))
         {
-            return l_bytesUpdatedOnDbus;
+            l_recordName = std::get<0>(*l_ipzData);
+            l_interfaceName = constants::ipzVpdInf + l_recordName;
+            l_propertyName = std::get<1>(*l_ipzData);
+
+            // Read the keyword's value from hardware to write the same on
+            // D-bus.
+            l_keywordValue = readKeyword(
```

Comment:
Need to handle exception thrown from this API?
Better increase the scope of try/catch?

Reply:
done